### PR TITLE
New Building explosions and SaberCut change

### PR DESCRIPTION
**Pull request by trempler**
(Defence/Support Buildings will need work later also units)
Changed repair Intervall from 6 down to 4
Added new Tesla Weapon to SaberCut.

### CODE REVIEW

**Review by CDVoidwalker — CHANGES_REQUESTED** [review]
Looks good to me apart from renaming SabercutLaser to SabercutLaserNew

File: rules/defaults.yaml [hunk 1]
```diff
@@ -433,9 +433,19 @@
 	ExternalCapturable:
 		CaptureCompleteTime: 7
 	ExternalCapturableBar:
+	GrantConditionOnDamageState@smoke:
+		Condition: damaged
+		ValidDamageStates: Critical
```
Comment (GraionDilach) [inline rules/defaults.yaml]: I'd consider enabling this on Heavy already. TS worked that way.

File: rules/defaults.yaml [hunk 2]
```diff
@@ -468,22 +478,46 @@
...
 	ThrowsShrapnel:
 		Weapons: LargeDebris
-		Pieces: 2, 20
-		Range: 2c848, 7c73
+		Pieces: 4, 15
+		Range: 3c512, 5c73
+	ThrowsShrapnel@SmallDebris:
+		Weapons: SmallDebris
+		Pieces: 10, 30
+		Range: 3c000, 6c000
```
Comment (GraionDilach) [inline rules/defaults.yaml]: Redundant zeroes (3c0 is enough)

File: rules/defaults.yaml [hunk 3]
```diff
@@ -468,22 +478,46 @@
 		MaxHeightDelta: 3
 	ThrowsShrapnel:
 		Weapons: LargeDebris
-		Pieces: 2, 20
-		Range: 2c848, 7c73
+		Pieces: 4, 15
+		Range: 3c512, 5c73
+	ThrowsShrapnel@SmallDebris:
+		Weapons: SmallDebris
+		Pieces: 10, 30
```
Comment (GraionDilach) [inline rules/defaults.yaml:485]: The amount worries me. CD is already a performance hog due to the sheer amount of debris spawned.

File: rules/palettes.yaml [hunk 4]
```diff
@@ -20,6 +20,30 @@
 	PaletteFromCurrentTileset@decoration:
 		Name: terraindecoration
 		ShadowIndex: 1
+	PaletteFromFile@apolblue:
+		Name: apolblue
+		Filename: apolblue.pal
+		Alpha: 0.8
```
Comment (GraionDilach) [inline rules/palettes.yaml]: These values won't work. PaletteFromFile doesn't support alpha values.

You need to create a PaletteFromPaletteWithAlpha derivate if you'd really intend to use alpha. Pretty sure examples already exist in this file.

File: weapons/cabal-weapons.yaml [hunk 5]
```diff
@@ -697,33 +697,46 @@ TarantulaLaserAA:
...
-		Color: 0064FF
-		Width: 0c57
-		BeamDuration: 3
-		ZOffset: 2047
-		UsePlayerColor: false
-		SecondaryBeam: true
-		SecondaryBeamWidth: 0c28
-		SecondaryBeamZOffset: 2047
-		SecondaryBeamColor: 00C8FF
+	Charges: yes
```
Comment (GraionDilach) [inline weapons/cabal-weapons.yaml]: This tag doesn't exist.

File: weapons/shared-weapons.yaml [hunk 6]
```diff
@@ -293,41 +293,122 @@ SmallDebris:
...
 	Inherits: SmallDebrisFast
+	Range: 2c512
 	Projectile: BulletAS
-		Speed: 0c120, 0c250
+		Speed: 0c170
 		Image: dbrislg
-		TrailImage: durasmoke
+		TrailImage: exp33fr
 		TrailPalette: apolra25alpha
+    Warhead@1Eff: CreateEffect
```
Comment (GraionDilach) [inline weapons/shared-weapons.yaml]: Messed up indentation.

File: weapons/shared-weapons.yaml [hunk 7]
```diff
@@ -293,41 +293,122 @@ SmallDebris:
...
+
+LargeDebrisFastcabal:
+	Inherits: SmallDebrisFast
+	Range: 0c750
+	Projectile: BulletAS
+		Speed: 0c250
+		Image: dbrislg
+		TrailImage: blueexpl01
+		TrailPalette: effect50alpha
+    Warhead@1Eff: CreateEffect
```
Comment (GraionDilach) [inline weapons/shared-weapons.yaml]: Messed up indentation.

File: weapons/shared-weapons.yaml [hunk 8]
```diff
@@ -419,9 +508,14 @@ Buildingexplosionmedium:
 
 Buildingexplosionmediumcabal:
 	Inherits: Buildingexplosionmedium
+	Warhead@1Eff: CreateEffect
+		Explosions: blueexpl01, blueexpl02, blueexpl03, blueexpl04
+		ExplosionPalette: effect25alpha
 	Warhead@3Eff: CreateEffect
 		Explosions: cabalring
 		ExplosionPalette: effect
+	Warhead@6Eff: CreateEffect
```
Comment (GraionDilach) [inline weapons/shared-weapons.yaml:517]: This warhead lacks an explosion?!

Comment (GraionDilach) [reply weapons/shared-weapons.yaml:517]: Ah, it's inherited, nevermind this one then.

File: rules/palettes.yaml (hunk 4 above)
Comment (GraionDilach) [reply rules/palettes.yaml]: Note that this applies to all new palette entries.

File: rules/defaults.yaml (hunk 1 above)
Comment (jrb0001) [reply rules/defaults.yaml]: Fixed.

File: rules/defaults.yaml (hunk 2 above)
Comment (jrb0001) [reply rules/defaults.yaml]: Fixed.

File: rules/defaults.yaml (hunk 3 above)
Comment (jrb0001) [reply rules/defaults.yaml:485]: Needs feedback from @trempler.

File: rules/palettes.yaml (hunk 4 above)
Comment (jrb0001) [reply rules/palettes.yaml]: Also applies to existing `PaletteFromFile@alpha2:`. Needs feedback from @trempler.

File: weapons/cabal-weapons.yaml (hunk 5 above)
Comment (jrb0001) [reply weapons/cabal-weapons.yaml]: Fixed.

File: weapons/shared-weapons.yaml (hunk 6 above)
Comment (jrb0001) [reply weapons/shared-weapons.yaml]: Fixed.

File: weapons/shared-weapons.yaml (hunk 7 above)
Comment (jrb0001) [reply weapons/shared-weapons.yaml]: Fixed.

**Comment by jrb0001** [comment]
@CDVoidwalker fixed your point.

File: rules/defaults.yaml (hunk 3 above)
Comment (trempler) [reply rules/defaults.yaml:485]: can be adjusted, runs fine here but depends on machine I guess.

File: rules/palettes.yaml (hunk 4 above)
Comment (trempler) [reply rules/palettes.yaml]: Will check coming week or sunday, did not noticed this wont work. maybe not needed for that anyway. So rather removing the Alpha value.

Comment (jrb0001) [reply rules/palettes.yaml]: Fixed.

**Review by CDVoidwalker — APPROVED** [review]

**Pull request merged**